File: py/problem4civ.py
import matplotlib.pyplot as plt
import numpy

f = lambda x: (2*x**3 + 3*x - 1)*numpy.cos(x) - x

xs = numpy.linspace(-1, 1, 400)
y = f(xs)
plt.plot(xs, y)
plt.axhline(0, color="black", linewidth=0.5)
plt.show()

File: m/problem4civ.m
clc; close all; clear variables;

f = @(x) (2*x.^3 + 3*x - 1).*cos(x) - x;

xs = linspace(-1, 1, 400);
y = f(xs);
figure;
plot(xs, y);
hold on;
yline(0, 'Color', 'black', 'LineWidth', 0.5);
%